content/i18n/en/docs/language/detailed-rules/typecobol-functions-syntax.md

```diff
@@ -0,0 +1,326 @@
+		END-CALL?
+```
+
+#### Function invocation syntax
+```cobol
+		<function-name> ( <arguments>? )
+```
+
+#### \_INPUT\_ clause syntax
+```cobol
+		INPUT < <SENDING MODE clause>? <parameter-name> >+
+```
+#### \_SHARING MODE\_ clause syntax
+```cobol
+		BY? <CONTENT | REFERENCE | VALUE>
+```
+
+#### \_INOUT\_ clause syntax
+```cobol
+		INOUT <parameter-name>+
+```
+
+#### \_OUTPUT\_ clause syntax
+```cobol
+		OUTPUT <parameter-name>+
+```
+
+<a name="call"/>
+
+# Caller / How to call a function or procedure
+
+## Procedure
+```cobol
+		CALL procedure-name INPUT      inputParam1 ... inputParamN
+							IN-OUT     inoutParam1 ... inoutParamN
+							OUTPUT     outParam1   ... outParamN
+		end-call
+		if procedure-name::error-code = '0000'
+		   ... return OK
+		else 
+		   ... return KO, handle error
+		end-if
+```
+
+## Functions
+```cobol
+		FUNCTION function-name (param1, param2, param3, ... paramN)
+```
+This syntax is the same as the syntax for COBOL intrinsic functions and can be used in the same contexts. In particular:
+  * __TCRFUN\_TCFUNC\_WHERE\_COBOL85FUNC__\_ a custom function can be called as part of any statement where an COBOL 85 intrinsic function call would be acceptable. **TODO: see if we can allow intrinsic functions in more contexts than COBOL 85.**
+  * __TCRFUN\_OPTIONAL\_FUNCTION__ the `FUNCTION` keyword is optional
+However, if there is any ambiguity about *function-name*, the function call must be qualified as described in [[Namespaces & imports|Namespaces]].
+
+### Alternate call for function
+__TCRFUN\_FUNC\_ALLOW\_CALL__ Functions can also be called like a procedure. This syntax make it easier to test the error-code.
+```cobol
+		CALL function-name INPUT      inputParam1, ... inputParamN
+						   RETURNING  returningParameter 
+		end-call
+		if function-name::error-code = '0000'
+		   ... return OK
+		else 
+		   ... return KO, handle error
+		end-if
+```
+
+## Common rules for functions and procedures calls
+
+* Rule that need to be validated \_TCRFUN\_FUNC\_COMMA\_SEPARATOR  You need to separate parameters with a comma (`,`).\_
+
+* __TCRFUN\_CALL\_PARAMETER\_ORDER__
+  During a procedure invocation, parameters are always used in the same order they are declared in the corresponding procedure:
+  1. First, all `INPUT` parameters (if any)
+  2. Second, all `INOUT` parameters (if any)
+  3. Third, all `OUTPUT` parameters (if any)
+
+
+* __TCRFUN\_INPUT\_BY__   `INPUT` parameters can be passed by content, by reference or by value.
+  This distinction is made for a given parameter using the \_SHARING MODE\_ clause.
+  If the \_SHARING MODE\_ clause is omitted for a given parameter, the \_SHARING MODE\_ of the parameter
+  declared immediately before on the current `INPUT` clause is used. 
+
+  If there is no previous parameter immediately before on the current `INPUT` clause, then `BY REFERENCE` is assumed. 
+
+  For example, in the following declaration:
+** `i1`, `i4`, `i5`, `o1`, `o2` are passed `BY REFERENCE`
```

Comment:
Missing closing ** ?

Reply:
I really don't know, this is the same situation at https://github.com/TypeCobolTeam/TypeCobol/wiki/TypeCobolFunctionsSyntax